src/main/scala/com/wavesplatform/state2/StateWriter.scala

```diff
@@ -95,8 +96,49 @@ class StateWriterImpl(p: StateStorage, synchronizationToken: ReentrantReadWriteL
     }
 
     measureSizeLog("contracts")(blockDiff.txsDiff.contracts) {
-      _.foreach { case (name, (status, account, content)) =>
-        sp().contracts.put(name, (status, account.bytes, content))
+      _.foreach { case (id, (h, txId, contract, _)) =>
+        sp().contracts.put(id, (h, txId, contract.bytes.arr))
+      }
+    }
+
+    measureSizeLog("accountContractIds")(blockDiff.txsDiff.accountContractIds) {
+      _.foreach { case (acc, ctIds) =>
+        val startIdxShift = sp().accountContractsLengths.getOrDefault(acc.bytes, 0)
+        ctIds.reverse.foldLeft(startIdxShift) { case (shift, ctId) =>
+          sp().accountContractIds.put(accountIndexKey(acc, shift), ctId)
+          shift + 1
+        }
+        sp().accountContractsLengths.put(acc.bytes, startIdxShift + ctIds.length)
+      }
+    }
+
+    measureSizeLog("contractDB")(blockDiff.txsDiff.contractDB) {
+      _.foreach { case (id, contractData) =>
+        sp().contractDB.put(id, contractData)
+      }
+    }
+
+    measureSizeLog("contractTokens")(blockDiff.txsDiff.contractTokens) {
+      _.foreach { case (id, tokenNum) =>
+        Option(sp().contractTokens.get(id)) match {
+          case Some(num) => sp().contractTokens.put(id, num + tokenNum)
+          case None => sp().contractTokens.put(id, tokenNum)
+        }
+      }
+    }
+
+    measureSizeLog("tokenDB")(blockDiff.txsDiff.tokenDB) {
+      _.foreach { case (id, tokeninfo) =>
+        Option(sp().tokenDB.put(id, tokeninfo))
+      }
+    }
+
+    measureSizeLog("tokenAccountBalance")(blockDiff.txsDiff.tokenAccountBalance) {
+      _.foreach { case (id, balance) =>
+        Option(sp().tokenAccountBalance.get(id)) match {
+          case Some(bl) => sp().tokenAccountBalance.put(id, safeSum(bl, balance))
+          case None => sp().tokenAccountBalance.put(id, balance)
+        }
```

Comment:
@Icermli I think you may follow the idea of line 52.
`val updateBalance = *`, and then `sp.tokenAccountBalance.put(id, updateBalance)`